File: 源代码/后端/his_pharmacy_mag/src/main/java/com/pharmacy/management/controller/LoginController.java
package com.pharmacy.management.controller;

import com.pharmacy.management.bean.User;
import com.pharmacy.management.result.LoginResult;
import com.pharmacy.management.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.util.HtmlUtils;

/**
 * @author 徐奥飞
 * @date 2019-11-5 8:43
 */

@Controller
public class LoginController {

    @Autowired
    UserService userService;

    @CrossOrigin // 解决跨域问题
    @PostMapping(value = "/api/login")
    @ResponseBody
    public LoginResult login(@RequestBody User requestUser) {
        // 对 html 标签进行转义，防止 XSS 攻击
        String username = requestUser.getUsername();
        username = HtmlUtils.htmlEscape(username);

        User user = userService.getByName(username);
        if (user == null) {
            return new LoginResult(400, "此用户名不存在！");
        } else if (user.getPassword().equals(requestUser.getPassword())) {
            return new LoginResult(200);
        } else {
            return new LoginResult(400, "密码错误！");
        }
    }

}

File: 源代码/后端/his_pharmacy_mag/src/main/java/com/pharmacy/management/controller/LoginController.kt
package com.pharmacy.management.controller

import com.pharmacy.management.api.CommonResult
import com.pharmacy.management.entity.User
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController
/**
 * @author 徐奥飞
 * @date 2019-11-5 9:30
 */
@RestController //此注解是 @ResponseBody 和 @Controller 的组合注解，可返回一个 JSON
class LoginController {
    @RequestMapping(value = ["/admin/login"], method = [RequestMethod.POST])
    fun admin(@RequestBody user: User): CommonResult<*> {
        return if (user.username == "admin" && user.password == "123456") {
            CommonResult.success("admin")
        } else {
            CommonResult.validateFailed()
        }
    }
}